### ISSUE TITLE
Historical Logs

### ISSUE THREAD

**Issue by agn507** [springbok/MLM2PRO-GSPro-Connector#161]
I was doing some testing and wanted to verify if the logs showed online/offline mode calculation adjustments. I switched offline mode to yes which let me know I needed to restart the connector. What I didn't realize is I'd lose my logs from the previous session which is what I was comparing against. Can the connector logs in appdata/logs/mlm2pro-gspro-connect.txt get a date/time appended so we can go back and review if needed. Could certainly be an optional thing and maybe allow for a max number of logs to save.

**Comment by wonder99** [comment]
I also deal with multiple log files, but I just open it up, and save it under a new file name before relaunching.  Is that OK for you?  It makes comparing the files much easier too.
I don't want to keep appending to the log, as it will get huge quickly.

**Comment by agn507** [comment]
Yeah that could certainly help if I remember to. What I was more suggesting is that each time the connector app opens instead of starting fresh and clearing out the previous log the current log would just be mlm2pro-gspro-connect-2024-12-14@07:52PM.txt. That way when you close your session the historical data is there. When you open again you'll just get a fresh logs file for the current date/time. 

No worries if this is more work than it's worth. Can certainly make a manual backup on close.

**Comment by springbok** [comment]
I could add a new setting that you can turn on/off to keep a log history, not everyone needs this and to conserve space it would be off by default.I can highlight any offline corrections in the log more clearly in the log file as well.Sent from my iPhoneOn 15 Dec 2024, at 11:53 am, Anthony Neumann ***@***.***> wrote:﻿
Yeah that could certainly help if I remember to. What I was more suggesting is that each time the connector app opens instead of starting fresh and clearing out the previous log the current log would just be ***@***.***:52PM.txt. That way when you close your session the historical data is there. When you open again you'll just get a fresh logs file for the current date/time.
No worries if this is more work than it's worth. Can certainly make a manual backup on close.

**Comment by springbok** [comment]
The following change will be in the next release:

New setting added 'Keep Log History', when set to 'Yes' each log file will be created with a date & time in the filename

**Comment by springbok** [comment]
This feature has been added to https://github.com/springbok/MLM2PRO-GSPro-Connector/releases/tag/V1.04.20 which was released today.
